generator.go

```diff
@@ -181,6 +182,9 @@ func (g *Generator) Generate() (GenerateResult, error) {
 }
 
 func satisfies(images []platform.RunImageForExport, imageName string) bool {
+	if imageName == "" {
+		return true
+	}
```

Comment:
This covers the case where we didn't switch the run image, so the original run image still satisfies the constraints. But we might be able to move this check to a place where its meaning is more obvious